File: app/src/main/java/com/jackyli/androidarchitecture/ui/login/LoginActivity.java
package com.jackyli.androidarchitecture.ui.login;

import android.view.View;
import android.widget.Toast;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModelProvider;

import com.jackyli.androidarchitecture.R;
import com.jackyli.androidarchitecture.base.BaseActivity;
import com.jackyli.androidarchitecture.databinding.ActivityLoginBinding;
import com.jackyli.androidarchitecture.model.UserInfo;
import com.jackyli.androidarchitecture.ui.vms.LoginViewModel;

import java.util.List;

public class LoginActivity extends BaseActivity<ActivityLoginBinding> {

    private LoginViewModel loginViewModel;

    @Override
    public void initView() {
        binding.setClick(this);
        loginViewModel = new ViewModelProvider(this).get(LoginViewModel.class);
        addObserver();
    }

    @Override
    public ActivityLoginBinding initLayout() {
        return DataBindingUtil.setContentView(this, R.layout.activity_login);  //绑定视图
    }

    public void click(View view) {
        loginViewModel.login(binding.username.getText().toString(), binding.password.getText().toString());
    }

    private void addObserver() {
        loginViewModel.getLoginResult().observe(this, this::showLoginResult);
        loginViewModel.getUser().observe(this, this::showUserPass);
    }

    private void showLoginResult(Boolean aBoolean) {
        if (aBoolean) {
            Toast.makeText(LoginActivity.this, "登录成功", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(LoginActivity.this, "登录失败", Toast.LENGTH_SHORT).show();
        }
    }

    private void showUserPass(List<UserInfo> userInfoList) {
        if (userInfoList != null && userInfoList.size() > 0) {
            UserInfo userInfo = userInfoList.get(0);
            binding.username.setText(userInfo.getUserName());
            binding.password.setText(userInfo.getPassWord());
        }
    }
}

File: app/src/main/java/com/jackyli/androidarchitecture/ui/login/LoginActivity.kt
package com.jackyli.androidarchitecture.ui.login

import android.view.View
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.databinding.Observable
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.jackyli.androidarchitecture.MvvmApplication
import com.jackyli.androidarchitecture.R
import com.jackyli.androidarchitecture.base.BaseActivity
import com.jackyli.androidarchitecture.databinding.ActivityLoginBinding
import com.jackyli.androidarchitecture.ui.vms.LoginViewModel
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : BaseActivity<ActivityLoginBinding>() {

    private lateinit var loginViewModel: LoginViewModel

    override fun initLayout(): ActivityLoginBinding {
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        return binding
    }

    override fun initView() {
        binding.click = this
        loginViewModel = ViewModelProvider(this).get(LoginViewModel::class.java)
        addObserver()
    }

    private fun addObserver() {
        loginViewModel.getLoginResult().observe(this, Observer {
            if (it) {
                Toast.makeText(MvvmApplication.getContext(), "登录成功", Toast.LENGTH_SHORT).show();
            } else {
                Toast.makeText(MvvmApplication.getContext(), "登录失败", Toast.LENGTH_SHORT).show();
            }
        })
    }

    fun click(v: View) {
        loginViewModel.login(username.text.toString(), password.text.toString())
    }
}